lib/intercept.js

```diff
@@ -124,6 +124,7 @@ function remove(interceptor) {
 }
 
 function removeAll() {
+  common.removeAllTimeouts()
```

Comment:
I'm all for adding this functionality. 👍Thank you for picking this up! 💯 

Though I'm not sure I think it belongs in the call to `clearAll()`. That function's job is to remove interceptors from the list. In contrast, what's happening here is that we're cancelling playback of an interceptor that has already matched and is in process.

I'd suggest exporting this as a new function, maybe calling it `nock.abortRequests()` or `nock.abortPendingRequests()`. The test name in `common` seems fine, "timeouts" being the implementation.

In [RFC-001](https://github.com/nock/nock/blob/master/rfcs/rfc-001.md) I advocated creating a new lifecycle method to reset nock to its initial state.

I am not sure whether aborting current requests should be added to that reset. There is a bit of discussion in #1118 about Mocha leaving the process running until the pending request flushes, though I think I could use a bit more clarification about the use case and why aborting the requests is what's desirable. (It's not the behavior of real HTTP to abort pending requests.) The reason Mocha doesn't quit the process anymore is to not paper over sloppy cleanup. I'm happy to let people force automatic cleanup if they want it, though people may prefer to keep things cleaned up on their own. Mocha's choice isn't everyone's cup of tea but it's arguably a good default.

That reset function doesn't exist yet; as of now you need to run `nock.restore(); nock.cleanAll(); nock.enableNetConnect(); nock.activate()` and when we add `nock.reset()` I'd suggest we stick with those four  calls to start. And that developers call `nock.abortPendingRequests()` followed by `nock.reset()` if they're experiencing this problem. We can gather feedback on the use cases over time before deciding to include that reset function by default.

Reply:
This commit was created to close the original bug which complained about cleanAll not removing pending request. However, I agree with you that cleanAll should stay as is and a new method should be exposed I am partial to"abortPendingRequests".  I dont see a current issue raised for nock.reset() be happy to work on that too. Sounds like it could be nocked out quickly.

Reply:
Cant clarify why this behavior is desirable, I personally am confused why someone needs it as well. I was just looking to contribute to your project because i was able to take the mocking approach and apply to one of my own projects. My way of saying thank you!

Reply:
Hey @TLPcoder I am really happy to have your help on the library. Nock is several years old – definitely a legacy project – and we've done a good job of getting it in shape to work on. @mastermatt has also done a lot of great work to clarify the ambiguous parts of the API. At the same time it still has a huge API surface, mostly because of the inclusionist way it was maintained, which was to merge many PRs that handled everyone's various niche use cases. A goal is to pare that back where possible, and I do want to be careful and thoughtful about expanding the API surface. Re the new lifecycle API, what I'd suggest is that we start by reviving #1441. Basically that PR got really big, and should be broken out into four separate RFCs. (The analysis of lifecycle methods RFC-001 came from the preamble that preceded the four proposals, via #1474.) The comments on the proposals also need to be incorporated.

In part the goal of the RFCs is to generate comments; and in part it's to have a thoughtful analysis and discussion that folks can read to understand the rationale as they migrate to the new APIs.

Would love to have your help on this!

Reply:
I think guys did a good job refactoring the code base and moving everything over to ES6.

Be happy to help where I can.